vmm/src/lib.rs

```diff
@@ -1666,6 +1621,54 @@ impl Vmm {
     }
 }
 
+/// Configures logger using a `LoggerConfig`.
+pub fn configure_logger(logger_config: LoggerConfig, firecracker_version: &str) -> UserResult {
+    LOGGER.set_level(match logger_config.level {
+        LoggerLevel::Error => Level::Error,
+        LoggerLevel::Warning => Level::Warn,
+        LoggerLevel::Info => Level::Info,
+        LoggerLevel::Debug => Level::Debug,
+    });
+
+    LOGGER.set_include_origin(logger_config.show_log_origin, logger_config.show_log_origin);
+    LOGGER.set_include_level(logger_config.show_level);
+
+    #[cfg(target_arch = "aarch64")]
+    let options: &Vec<Value> = &vec![];
+
+    #[cfg(target_arch = "x86_64")]
+    let options = logger_config.options.as_array().unwrap();
```

Comment:
This is not related to this PR. The logger has architecture specific behavior because the logger configuration is the place where you set the dirty log capability to on or off. logger_config.options is an array that can zero elements (dirty log is not enabled) or one element which must be LogDirtyPages.

Since right now the KVM_GET_DIRTY_LOG wrapper only exists on x86_64 the options are only valid on x86_64, hence the arch-specific code.